### ISSUE TITLE
Nothing is displayed when value is programatically set and getOptionValue is used

### ISSUE THREAD

**Issue by klinki** [JedWatson/react-select#4973]
When `getOptionValue` is used, `value` or `defaultValue` still require using full object and don't respect the `getOptionValue` result.

```typescript
const allItems: Item[] = [
  { id: 1, name: "Tesla" },
  { id: 2, name: "BMW" },
  { id: 3, name: "Audi" },
  { id: 4, name: "Mazda" }
];

  const getLabel = (item: Item) => item.name;
  const getValue = (item: Item) => item.id.toString();
```

And when I use 
```jsx
        <Select
          getOptionLabel={getLabel}
          getOptionValue={getValue}
          options={allItems}
          defaultValue={allItems[0].id.toString()}
        />
```

it doesn't work and doesn't select the proper value. Neither does using `value` property. But if I use `defaultValue={allItems[0]}` it works - so it seems it doesn't respect the usage of `getOptionValue`.

Normally, I would use something like `allItems.find(item => item.id.toString() === selectedId)`, but that is not possible to use in `AsyncSelect`, since I don't have access to loaded options.

(I'm trying to use react-select with react-hook-forms and I have noticed these issues while working on this integration. Here is more detailed example: https://codesandbox.io/s/admiring-monad-tq2er?file=/src/App.tsx )

**Comment by Rall3n** [comment]
Hello @klinki,

the values for the `value` and `defaultValue` props should have the same schema as a single option in your list of options, as both function essentially independently from the `options` prop. In other words: The value does not have to be in the option list.

I can only theorize as to why this is: Imagine having an asynchronous `Select` component with options from an API. The options are never sent as a whole, but as a subset of the available options filtered by the search terms. You can not guarantee that the options contain the current value.

**Comment by klinki** [comment]
> Hello @klinki,
> 
> the values for the `value` and `defaultValue` props should have the same schema as a single option in your list of options, as both function essentially independently from the `options` prop. In other words: The value does not have to be in the option list.

Well, I'm not so sure about this one. Shouldn't they have the same schema as output of `getOptionValue` function call?

Honestly, I don't care that much if the value is or is not in the list (my example with array was meant as a workaround for this issue, not the solution). I just need it to respect the `getOptionValue` function, because I want to use `react-select` in the form and integrate it with `react-hook-forms`.

**Comment by tm-noelzubin** [comment]
Agreed. It would be nice if react select allowed api similar to this. 
```
const options = [ { value: 1, label: "one" } , { value: 2, label: "two" }];

const [value, setValue] = useState(1); // value is only number

<Select 
  options={options}
  value={value}
  onChange={val => setValue(val)}
  getOptionValue={opt => opt.value}
/>
```

**Comment by tm-noelzubin** [comment]
I'm using this for now.
```
import Select from 'react-select';

const SimpleSelect = ({ value, onChange, options, ...props }) => {
	const selectVal = options.find((opt) => opt.value === value);
	const onSelectChange = ({ value }) => onChange(value);

	return (
		<Select
			{...props}
			options={options}
			value={selectVal}
			onChange={onSelectChange}
		/>
	);
};

export default SimpleSelect;
```

**Comment by klinki** [comment]
That's nice, but impossible to use with `AsyncSelect`

**Comment by Stanislav-Lapata** [comment]
Thanks for your great project

> That's nice, but impossible to use with `AsyncSelect`

In my opinion, value props should take 
```javascript
value={(valuesFromPromise) => valuesFromPromise.find(value => value.id === idFromUseState)}
```
as example
but I can be wrong

**Comment by wallace-sf** [comment]
Any workaround on this?

**Comment by wallace-sf** [comment]
Any workaround on this? It's impossible to set a default Value as a string.
